src/phases/execution/ClusteredCollection.yml

```diff
@@ -136,15 +135,41 @@ Actors:
           OperationCommand:
             Filter: {a: {^FastRandomString: {length: 6, alphabet: "0123456789"}}}
 
+# Phase 6: Range lookup with both primary and secondary key
+- Name: RangeLookupByPrimaryAndSecondaryIndexes
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [6]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {
+              _id: {
+                $gte: {^Join: {array: [{ ^FormatString: {"format": "%07d", "withArgs": [{^RandomInt: {min: 1, max: *docCount}}]}}, *PaddingGenerator]}}
+              },
+              a: {
+                $gte: {^Join: {array: [
+                  {^Choose: {from: ["49", "51"]}},
```

Comment:
Is there any specific reason for this prefixes?

Reply:
The idea is to have index scan that scans over roughly the half of all documents.
I also want _id predicate to be similar: so not random int 1 - docsCount, but maybe something like 49 * docsCount / 100 - 51 * docsCount / 100. But I don't know how to implement it yet.

Currently I am testing if this works and get back to this idea later.
Maybe I will just scrap this idea and make range on a also random. 

Multiplanner should be able to pick correct index to scan in any case.

Reply:
Not sure if it will help, but you can do YAML arithmetic (preprocessed) with `^NumExpr`.

Reply:
Thanks. That is exactly what I was looking for.

Currently I decided to just pick ranges at random. It makes one of the two indexes a clear winner during multiplanner, which should be fine. I was over-complicating stuff.

Maybe it is a good idea to add two separate phases: one will have _id index be better fit for the query and the other will have secondary index clearly better.